File: frontend/src/types/feedback.ts
export interface AppFeedbackCreate {
    subject: string;
    message: string;
    rating?: number;
    category?: string;
}

export interface CourseFeedbackCreate {
    course_id: number;
    rating: number;
    review_text?: string;
}

export interface InstructorFeedbackCreate {
    instructor_id: number;
    rating: number;
    comments?: string;
}

export interface FeedbackResponse {
    id: number;
    user_id?: number;
    subject?: string;
    rating?: number;
    message?: string;
    review_text?: string;
    comments?: string;
    category?: string;
    status?: string;
    response?: string;
    created_at?: string;
    user?: {
        id: number;
        full_name: string;
        email: string;
    };
}

File: backend/app/schemas/feedback.py
from typing import Optional, List
from pydantic import BaseModel
from datetime import datetime


class FeedbackBase(BaseModel):
    subject: str
    message: str
    rating: Optional[int] = None
    category: str = "general"


class AppFeedbackCreate(FeedbackBase):
    pass


class CourseFeedbackCreate(BaseModel):
    course_id: int
    rating: int
    review_text: Optional[str] = None


class InstructorFeedbackCreate(BaseModel):
    instructor_id: int
    rating: int
    comments: Optional[str] = None


class UserBasic(BaseModel):
    id: int
    full_name: str
    email: str
    
    class Config:
        from_attributes = True


class AppFeedbackResponse(BaseModel):
    id: int
    user_id: Optional[int] = None
    subject: str
    message: str
    rating: Optional[int] = None
    category: str
    status: str
    response: Optional[str] = None
    created_at: Optional[datetime] = None
    user: Optional[UserBasic] = None
    
    class Config:
        from_attributes = True


class CourseFeedbackResponse(BaseModel):
    id: int
    user_id: Optional[int] = None
    course_id: int
    rating: int
    review_text: Optional[str] = None
    created_at: Optional[datetime] = None
    user: Optional[UserBasic] = None
    
    class Config:
        from_attributes = True


class InstructorFeedbackResponse(BaseModel):
    id: int
    user_id: Optional[int] = None
    instructor_id: int
    rating: int
    comments: Optional[str] = None
    created_at: Optional[datetime] = None
    user: Optional[UserBasic] = None
    
    class Config:
        from_attributes = True


# Legacy response for backward compatibility
class FeedbackResponse(BaseModel):
    id: int
    user_id: Optional[int] = None
    rating: Optional[int] = None
    message: Optional[str] = None
    review_text: Optional[str] = None
    comments: Optional[str] = None
    created_at: Optional[datetime] = None
    user: Optional[UserBasic] = None
    
    class Config:
        from_attributes = True
